src/json/json_schema.rs

```diff
@@ -11,6 +11,7 @@ pub enum JsonSchema {
   Obj(HashMap<String, JsonSchema>),
   Arr(Box<JsonSchema>),
   Opt(Box<JsonSchema>),
+  Ref(String),
```

Comment:
How will Ref work if none of the JsonSchema types have any names?

Reply:
We use `Ref` only to refer to the `Obj` types which were referred again recursively. All the `Obj` types need to have a name for it to be defined recursively. So, there would always exist a type name. Is there any corner case I missed? Thanks.

Reply:
How about we clean this up a little bit as follows:

```rust
struct Name(String);
struct TypeDef(HashMap<Name, Definition>);
pub enum Definition {
  Obj(HashMap<String, Definition>),
  Arr(Box<Definition>),
  Opt(Box<Definition>),
  Str,
  Num,
  Bool,
  Ref(Name)
}

impl TypeDef {
  pub fn new() -> Self {todo!()}
  pub fn combine(self, other: TypeDefs) -> Self { todo!() }
  pub fn insert(self, def: Definition) -> Self {todo!()}
  pub fn validate(&self, value: &async_graphql::Value) -> Valid<(), &'static str> {todo!()}
}
```

Now validation is done on TypeDef which has access to set of types and can access by name.